daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/DaffodilTDMLDFDLProcessor.scala

```diff
@@ -160,9 +160,15 @@ final class TDMLDFDLProcessorFactory private (
     optRootName: Option[String],
     optRootNamespace: Option[String],
     tunables: Map[String, String]): TDML.CompileResult = {
-
-    val res = compileProcessor(schemaSource, useSerializedProcessor, optRootName, optRootNamespace)
-    res
+    if (schemaSource.isXSD) {
+      val res = compileProcessor(schemaSource, useSerializedProcessor, optRootName, optRootNamespace)
+      res
+    } else {
+      val dp = compiler.reload(schemaSource)
+      val diags = dp.getDiagnostics
+      Assert.invariant(diags.forall{! _.isError })
+      Right((diags, new DaffodilTDMLDFDLProcessor(dp)))
+    }
```

Comment:
You've added a test that takes us to line 167 and calls compiler.reload but the test forces that method to throw an InvalidParserException("The saved parser is only compatible with Daffodil " + savedVersion + ". Current version is " + curVersion) and we never get to lines 168 and 170, hence the coverage check warnings on these lines.  

You could write a test `testNonXSDModelPositiveGoodCompiledFile` which creates a parser, saves it to a temporary file (deleted automatically later), interpolates that temporary file's path into the test suite, and then runs the test suite like `testNonXSDModelNegativeBadCompiledFile` does except this test suite reloads the compiled parser successfully.

Reply:
I could do this, but I didn't because I deemed coverage of this particular functionality to be quite thorough in other places. This save/restore is used everywhere. Every TDML test, for example by default serializes and restores the processor before executing. So it's really just these two specific lines, and they are correct by inspection.